src/context/existingProject.tsx

```diff
@@ -0,0 +1,53 @@
+import React, {
+  createContext,
+  FC,
+  ReactNode,
+  useContext,
+  useState,
+} from "react";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { fetchProject } from "api/projects";
+
+interface ContextProps {
+  projectExists: boolean;
+  isProjectLoading: boolean;
+  setProject: (project: string) => void;
+}
+
+const initialState: ContextProps = {
+  projectExists: false,
+  isProjectLoading: false,
+  setProject: () => undefined,
+};
+
+export const ExistingProjectContext = createContext<ContextProps>(initialState);
+
+interface ProviderProps {
+  children: ReactNode;
+}
+
+export const ExistingProjectProvider: FC<ProviderProps> = ({ children }) => {
+  const [project, setProject] = useState("");
+
+  const { data, isLoading } = useQuery({
+    queryKey: [queryKeys.projects, project],
+    queryFn: () => fetchProject(project),
+  });
```

Comment:
Please configure useQuery to load once in case the url is 404, currently it loads the url three times.